fitbenchmarking/fitbenchmarking
# Refactor of table inheritance

#### Description of Work

Refactors the tables to be neater and more maintainable.

Most notably, this allows the tables to be sorted in a more flexible way.
This will be useful as more critereon are brought in and tables have more columns that need to be sorted.

#### Testing Instructions

1. The regressions tests will fail but it should only be due to a reordering of the elements.
   We will have to do them by hand.
2. Make some results with various jacobians, minimizers, softwares, etc. Do the tables look correct and match master?

Function: Does the change do what it's supposed to?
Tests: Does it pass? Is there adequate coverage for new code?
Style: Is the coding style consistent? Is anything overly confusing?
Documentation: Is there a suitable change to documentation for this change?
